File: resources/two_sum.ts
function two_sum(nums: number[], target: number): number[] {
  const numsMap: Map<number, number> = new Map()
  for (let idx: number = 0; idx < nums.length; idx++) {
    const right: number = nums[idx]
    const left: number = target - right
    if (numsMap.has(left)) return [numsMap.get(left), idx]
    numsMap.set(right, idx)
  }
  return []
}

const nums: number[] = [2, 11, 7, 15]
const target: number = 9

console.log(nums, target, two_sum(nums, target))

File: resources/two_sum.rs
pub struct Solution {}

use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut pairs = HashMap::with_capacity(nums.len());

        for (i, num) in nums.iter().enumerate() {
            match pairs.get(&(target - num)) {
                None => {
                    pairs.insert(num, i);
                }
                Some(sub_i) => return vec![*sub_i as i32, i as i32],
            }
        }
        vec![]
    }
}

fn main() {
    println!("hello baby");
}
